# Automobile Sales Analysis

## Business Objective

This project analyzes automobile sales performance, customer behavior, and revenue trends to identify key business insights and support data-driven decision-making.


## Analysis Targets

- Analyze sales and revenue trends over time.
- Identify top-performing products and product lines.
- Perform RFM customer segmentation.
- Detect potential customer churn.
- Analyze geographic sales distribution.
- Provide actionable business recommendations.


## Business Objective

## Revenue Performance Analysis

This project analyzes sales performance to identify key revenue drivers and segment customers based on purchasing behavior, supporting data-driven decisions to improve revenue growth and customer satisfaction.

## Analysis Targets

- Analyze sales and revenue trends over time.
- Identify top-performing products and product lines.
- Perform RFM (Recency, Frequency, Monetary) analysis.
- Identify top customers and their revenue contribution.
- Analyze geographic sales distribution.
- Provide actionable business recommendations.

# Environment Setup

In [ ]:
# Analysis library
import numpy as np
import pandas as pd

# visualization
import plotly.express as px
import plotly.graph_objects as go

print("Environment Version")
print(f"Numpy Version: {np.__version__}")
print(f"Pandas Version: {pd.__version__}")

Environment Version
Numpy Version: 2.0.2
Pandas Version: 2.2.2


# Data loading

## Data Loading

In [ ]:
# Load Dataset
df = pd.read_csv("Auto Sales data.csv")
print("Dataset Loaded successfully")

Dataset Loaded successfully


## Product Performance Analysis

In [ ]:
# Columns Descriptions
Column_info = {
  "ORDERNUMBER":"This column represents the unique identification number assigned to each order.",

  "QUANTITYORDERED":"It indicates the number of items ordered in each order.",

  "PRICEEACH":"This column specifies the price of each item in the order.",

  "ORDERLINENUMBER":"It represents the line number of each item within an order.",

  "SALES":"This column denotes the total sales amount for each order, which is calculated by multiplying the quantity ordered by the price of each item.",

  "ORDERDATE":"It denotes the date on which the order was placed.",

  "DAYS_SINCE_LASTORDER":	"This column represents the number of days that have passed since the last order for each customer. It can be used to analyze customer purchasing patterns.",

  "STATUS":	'It indicates the status of the order, such as "Shipped," "In Process," "Cancelled," "Disputed," "On Hold," or "Resolved.',

  "PRODUCTLINE":	"This column specifies the product line categories to which each item belongs.",

  "MSRP":	"It stands for Manufacturer's Suggested Retail Price and represents the suggested selling price for each item.",

  "PRODUCTCODE":	"This column represents the unique code assigned to each product.",

  "CUSTOMERNAME":	"It denotes the name of the customer who placed the order.",

  "PHONE":	"This column contains the contact phone number for the customer.",

  "ADDRESSLINE1":	"It represents the first line of the customer's address.",

  "CITY":	"This column specifies the city where the customer is located.",

  "POSTALCODE":	"It denotes the postal code or ZIP code associated with the customer's address.",

  "COUNTRY":	"This column indicates the country where the customer is located.",

  "CONTACTLASTNAME":	"It represents the last name of the contact person associated with the customer.",

  "CONTACTFIRSTNAME":	"This column denotes the first name of the contact person associated with the customer.",

  "DEALSIZE":	'It indicates the size of the deal or order, which are the categories "Small," "Medium," or "Large."'
}
pd.set_option('display.max_colwidth', None)

featrues_describtion = pd.DataFrame(Column_info.items(), columns=["Feature", "Description"])
featrues_describtion.set_index("Feature", inplace=True)
featrues_describtion

,Description
Feature,
ORDERNUMBER,This column represents the unique identification number assigned to each order.
QUANTITYORDERED,It indicates the number of items ordered in each order.
PRICEEACH,This column specifies the price of each item in the order.
ORDERLINENUMBER,It represents the line number of each item within an order.
SALES,"This column denotes the total sales amount for each order, which is calculated by multiplying the quantity ordered by the price of each item."
ORDERDATE,It denotes the date on which the order was placed.
DAYS_SINCE_LASTORDER,This column represents the number of days that have passed since the last order for each customer. It can be used to analyze customer purchasing patterns.
STATUS,"It indicates the status of the order, such as ""Shipped,"" ""In Process,"" ""Cancelled,"" ""Disputed,"" ""On Hold,"" or ""Resolved."
PRODUCTLINE,This column specifies the product line categories to which each item belongs.


# Structured Understanding

## Dataset Overview

In [ ]:
# Dataset information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2747 entries, 0 to 2746
Data columns (total 20 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   ORDERNUMBER           2747 non-null   int64  
 1   QUANTITYORDERED       2747 non-null   int64  
 2   PRICEEACH             2747 non-null   float64
 3   ORDERLINENUMBER       2747 non-null   int64  
 4   SALES                 2747 non-null   float64
 5   ORDERDATE             2747 non-null   object 
 6   DAYS_SINCE_LASTORDER  2747 non-null   int64  
 7   STATUS                2747 non-null   object 
 8   PRODUCTLINE           2747 non-null   object 
 9   MSRP                  2747 non-null   int64  
 10  PRODUCTCODE           2747 non-null   object 
 11  CUSTOMERNAME          2747 non-null   object 
 12  PHONE                 2747 non-null   object 
 13  ADDRESSLINE1          2747 non-null   object 
 14  CITY                  2747 non-null   object 
 15  POSTALCODE           

In [ ]:
# Missing Values OverView
missing = pd.DataFrame({"Missing Values":df.isna().sum()})
missing

,Missing Values
ORDERNUMBER,0
QUANTITYORDERED,0
PRICEEACH,0
ORDERLINENUMBER,0
SALES,0
ORDERDATE,0
DAYS_SINCE_LASTORDER,0
STATUS,0
PRODUCTLINE,0
MSRP,0


In [ ]:
# Duplicates Overview
duplicate_num = df.duplicated().sum()
print(f"The Number of Duplicates Rows: {duplicate_num}")

The Number of Duplicates Rows: 0


In [ ]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
ORDERNUMBER,2747.0,10259.761558,91.877521,10100.00,10181.000,10264.00,10334.500,10425.00
QUANTITYORDERED,2747.0,35.103021,9.762135,6.00,27.000,35.00,43.000,97.00
PRICEEACH,2747.0,101.098952,42.042549,26.88,68.745,95.55,127.100,252.87
ORDERLINENUMBER,2747.0,6.491081,4.230544,1.00,3.000,6.00,9.000,18.00
SALES,2747.0,3553.047583,1838.953901,482.13,2204.350,3184.80,4503.095,14082.80
DAYS_SINCE_LASTORDER,2747.0,1757.085912,819.280576,42.00,1077.000,1761.00,2436.500,3562.00
MSRP,2747.0,100.691664,40.114802,33.00,68.000,99.00,124.000,214.00


In [ ]:
# Numerical Features Summary
df.describe(include="O").T

,count,unique,top,freq
ORDERDATE,2747,246,14/11/2018,38
STATUS,2747,6,Shipped,2541
PRODUCTLINE,2747,7,Classic Cars,949
PRODUCTCODE,2747,109,S18_3232,51
CUSTOMERNAME,2747,89,Euro Shopping Channel,259
PHONE,2747,88,(91) 555 94 44,259
ADDRESSLINE1,2747,89,"C/ Moralzarzal, 86",259
CITY,2747,71,Madrid,304
POSTALCODE,2747,73,28034,259
COUNTRY,2747,19,USA,928


In [ ]:
df[["STATUS","PRODUCTLINE","DEALSIZE"]].nunique()

,0
STATUS,6
PRODUCTLINE,7
DEALSIZE,3


## Customer Analysis

In [ ]:
df[['ORDERNUMBER',"CUSTOMERNAME"]].nunique()

,0
ORDERNUMBER,298
CUSTOMERNAME,89


In [ ]:
df['PRODUCTLINE'].unique()

array(['Motorcycles', 'Classic Cars', 'Trucks and Buses', 'Vintage Cars',
       'Planes', 'Ships', 'Trains'], dtype=object)

In [ ]:
company_revenue = df["SALES"].sum()
print(f"Total Company Revenue: {round(company_revenue)}")
company_orders = df["ORDERNUMBER"].nunique()
print(f"Total Company Orders: {company_orders}")
company_customers = df["CUSTOMERNAME"].nunique()
print(f"Total Company Customers: {company_customers}")

Total Company Revenue: 9760222
Total Company Orders: 298
Total Company Customers: 89


## Final Business Understanding
* Structure:
Each row represents one product inside an order (not a full order).

* Business Flow:
Customer → Order → Product → Quantity × Price → Revenue → Time → Location

* Business View: The company sells different automobile-related products to many customers.
* The Company Genrated 9,760,221 Revenue From 298 orders made by 89 Customers

## Data cleaning

In [ ]:
df_clean = df.copy()
# Transform ORDER to Date type
df_clean["ORDERDATE"] = pd.to_datetime(df["ORDERDATE"],dayfirst=True )
df_clean['ORDERDATE'].dtype

dtype('<M8[ns]')

In [ ]:
# check Sales Calculations
uncorrect_sales = df[df['SALES']!=df['QUANTITYORDERED']*df['PRICEEACH']]["SALES"].size
print(f"The number of Uncorrected Sales: {uncorrect_sales} Record")

The number of Uncorrected Sales: 702 Record


Are there any discounts?

Did the price change at the time of sale?

Are there any taxes?

Is PRICEEACH not the final price?

# EDA

## Revenue Performance Analysis

In [ ]:
# Which Oeder Generate The Most Revenue
highest_revenue_orders = df_clean.groupby(["ORDERNUMBER"])["SALES"].sum().sort_values(ascending=False).reset_index(name="Order Revenue")
highest_revenue_orders.head(10)
print(f"The highest order revenue is {highest_revenue_orders["ORDERNUMBER"][0]}: {highest_revenue_orders["Order Revenue"][0]}$")

The highest order revenue is 10165: 77809.37$


## Data Cleaning & Preparation

In [ ]:
revenue_by_year = df_clean.resample("YE",on="ORDERDATE")["SALES"].sum().reset_index(name="Revenue")
revenue_by_year["pec_of_Total_revenue"]= (revenue_by_year["Revenue"]/company_revenue*100).round()
revenue_by_year["GROWTH"] = (revenue_by_year["Revenue"].pct_change() *100).fillna(0).round()

revenue_by_year

,ORDERDATE,Revenue,pec_of_Total_revenue,GROWTH
0,2018-12-31,3353014.06,34.0,0.0
1,2019-12-31,4669924.56,48.0,39.0
2,2020-12-31,1737283.09,18.0,-63.0


In [ ]:
revenue_by_month = df_clean.resample("ME",on="ORDERDATE")["SALES"].sum().reset_index(name="Revenue")
revenue_by_month["GROWTH"] = (revenue_by_month["Revenue"].pct_change() *100).fillna(0).round()

revenue_by_month = revenue_by_month.sort_values(by="Revenue",ascending=False)

In [ ]:
revenue_by_quarter = df_clean.resample("QE",on="ORDERDATE")["SALES"].sum().reset_index(name="Revenue")
revenue_by_quarter["GROTH"] = (revenue_by_quarter["Revenue"].pct_change() *100).fillna(0).round()
revenue_by_quarter

,ORDERDATE,Revenue,GROTH
0,2018-03-31,426399.11,0.0
1,2018-06-30,562365.22,32.0
2,2018-09-30,649514.54,15.0
3,2018-12-31,1714735.19,164.0
4,2019-03-31,809841.36,-53.0
5,2019-06-30,766260.73,-5.0
6,2019-09-30,1109396.27,45.0
7,2019-12-31,1984426.20,79.0
8,2020-03-31,1017788.74,-49.0
9,2020-06-30,719494.35,-29.0


* The analysis shows clear seasonality in revenue trends. Revenue increases significantly during Q4, especially in November, which represents the highest-performing month. In contrast, revenue tends to decline during the first half of the year.

## Product Performance Analysis

In [ ]:
top_product = (
    df.groupby(["PRODUCTLINE"])
    .agg({
        "QUANTITYORDERED": "sum",
        "SALES": "sum",
    })
    .sort_values(by="SALES", ascending=False)
    .rename(columns={
        "QUANTITYORDERED": "Total Quantity",
        "SALES": "Total Sales"


    })
    .reset_index()
)
top_product["PERCENTAGE OF SALES"] = (top_product["Total Sales"]/company_revenue*100).round()
top_product

,PRODUCTLINE,Total Quantity,Total Sales,PERCENTAGE OF SALES
0,Classic Cars,33373,3842868.54,39.0
1,Vintage Cars,20059,1806675.68,19.0
2,Trucks and Buses,10579,1111559.19,11.0
3,Motorcycles,11080,1103512.19,11.0
4,Planes,10636,969323.42,10.0
5,Ships,7989,700039.22,7.0
6,Trains,2712,226243.47,2.0


###The analysis shows that
 * revenue is highly concentrated in Classic Cars and Vintage Cars, which together contribute 58% of total revenue and have the highest sales volume.

 * Trains represent the weakest product line in both revenue, contributing only 2% of total sales. This category may require further business investigation regarding pricing, positioning, or customer demand.

 * Other product lines show relatively balanced performance but contribute less significantly compared to the leading categories.

## Customer Analysis

In [ ]:
customer_contribution = df_clean.groupby(["CUSTOMERNAME","ORDERDATE"])["SALES"].sum().sort_values(ascending=False).reset_index()
customer_contribution["ORDERDATE"] = pd.to_datetime(customer_contribution["ORDERDATE"])
customer_contribution["ORDERDATE"]= customer_contribution["ORDERDATE"].dt.year

customer_contribution = customer_contribution.pivot_table(index="CUSTOMERNAME",columns="ORDERDATE",values="SALES",aggfunc="sum").fillna(0).reset_index()

customer_contribution["total_sales"] = customer_contribution.iloc[:,1:].sum(axis=1)
customer_contribution["pec_of_total_sales"] = (customer_contribution["total_sales"]/company_revenue*100).round(2)
customer_contribution=customer_contribution.sort_values(by="total_sales",ascending=False).reset_index(drop=True)
customer_contribution.columns.name = None
customer_contribution.head(10)

,CUSTOMERNAME,2018,2019,2020,total_sales,pec_of_total_sales
0,Euro Shopping Channel,210227.58,375268.36,326798.17,912294.11,9.35
1,Mini Gifts Distributors Ltd.,185128.12,256474.26,213255.68,654858.06,6.71
2,"Australian Collectors, Co.",60135.84,140859.57,0.00,200995.41,2.06
3,Muscle Machine Inc,132778.24,64958.70,0.00,197736.94,2.03
4,La Rochelle Gifts,0.00,76722.81,103402.09,180124.90,1.85
5,"Dragon Souveniers, Ltd.",165686.20,3127.88,4175.60,172989.68,1.77
6,Land of Toys Inc.,25783.76,138285.68,0.00,164069.44,1.68
7,The Sharp Gifts Warehouse,0.00,64600.34,95409.93,160010.27,1.64
8,"AV Stores, Co.",51017.92,106789.89,0.00,157807.81,1.62
9,"Anna's Decorations, Ltd",88983.71,0.00,65012.42,153996.13,1.58


In [ ]:
check_column = [2018,	2019]
high_sustained_customer =customer_contribution[(customer_contribution[check_column] != 0).all(axis=1)].shape[0]
pct_sustained_customers = (high_sustained_customer/customer_contribution.shape[0])*100.


print(f"The Number of customers with sustained activity: {high_sustained_customer}")
print(f"The Percentage of customers with sustained activity: {round(pct_sustained_customers)}%")

The Number of customers with sustained activity: 59
The Percentage of customers with sustained activity: 66%


* Euro Shopping Channel and Mini Gifts Distributors Ltd. contribute nearly 17% of total revenue.

* Revenue is distributed across a large customer base and is not fully dependent on a few major customers.

* About 66% of customers show sustained purchasing activity across 2018 and 2019. The year 2020 was excluded from this comparison because the data is incomplete.

## RFM Customer Segmentation

## RFM  Analysis

### recency

In [ ]:
recency= df_clean.groupby(["CUSTOMERNAME"])["ORDERDATE"].max().reset_index()
recency["Days_science_lastorder"] = (df_clean["ORDERDATE"].max()-recency["ORDERDATE"]).dt.days
recency.drop(columns="ORDERDATE",inplace=True)
recency.sort_values(by="Days_science_lastorder",ascending=False,inplace=True,)
recency

,CUSTOMERNAME,Days_science_lastorder
29,"Double Decker Gift Stores, Ltd",495
87,West Coast Collectables Co.,488
70,Signal Collectibles Ltd.,476
25,Daedalus Designs Imports,465
21,Collectable Mini Designs Co.,460
...,...,...
53,Mini Gifts Distributors Ltd.,2
27,Diecast Classics Inc.,1
61,Petit Auto,1
43,La Rochelle Gifts,0


In [ ]:
R = df_clean.groupby(["CUSTOMERNAME","ORDERDATE"])["SALES"].sum().reset_index().sort_values(by=["CUSTOMERNAME","ORDERDATE"])

R["avg_purchase_interval"] = R.groupby("CUSTOMERNAME")["ORDERDATE"].diff().dt.days

R.fillna(0,inplace=True)

R = R.groupby("CUSTOMERNAME")["avg_purchase_interval"].mean().reset_index().round()
R

,CUSTOMERNAME,avg_purchase_interval
0,"AV Stores, Co.",203.0
1,Alpha Cognac,211.0
2,Amica Models & Co.,12.0
3,"Anna's Decorations, Ltd",136.0
4,Atelier graphique,185.0
...,...,...
84,"Vida Sport, Ltd",94.0
85,Vitachrome Inc.,221.0
86,"Volvo Model Replicas, Co",151.0
87,West Coast Collectables Co.,30.0


## Frequency

In [ ]:
F = df.groupby(["CUSTOMERNAME","ORDERNUMBER"])["SALES"].sum().reset_index()
F = F.groupby("CUSTOMERNAME").agg({"ORDERNUMBER":"count","SALES":"sum"}).reset_index()
F["AVG ORDER VALUE"] = (F["SALES"]/F["ORDERNUMBER"]).round()
F.rename(columns={"ORDERNUMBER":"Total Orders"},inplace=True)
F.drop(columns="SALES",inplace=True)
F

,CUSTOMERNAME,Total Orders,AVG ORDER VALUE
0,"AV Stores, Co.",3,52603.0
1,Alpha Cognac,3,23496.0
2,Amica Models & Co.,2,47059.0
3,"Anna's Decorations, Ltd",4,38499.0
4,Atelier graphique,3,8060.0
...,...,...,...
84,"Vida Sport, Ltd",2,58857.0
85,Vitachrome Inc.,3,29347.0
86,"Volvo Model Replicas, Co",4,18939.0
87,West Coast Collectables Co.,2,23042.0


* Most of our highest-value customers, in terms of revenue, are the most active, and their spending patterns per order fall into two categories: either high activity with low to medium spending, or lower activity with high spending.

In [ ]:
M= df.groupby(["CUSTOMERNAME"])["SALES"].sum().reset_index()
M

,CUSTOMERNAME,SALES
0,"AV Stores, Co.",157807.81
1,Alpha Cognac,70488.44
2,Amica Models & Co.,94117.26
3,"Anna's Decorations, Ltd",153996.13
4,Atelier graphique,24179.96
...,...,...
84,"Vida Sport, Ltd",117713.56
85,Vitachrome Inc.,88041.26
86,"Volvo Model Replicas, Co",75754.88
87,West Coast Collectables Co.,46084.64


###

### RFM Customer Segmentation & Churn Risk

In [ ]:
RFM =pd.merge(R,F,on="CUSTOMERNAME")
RFM =pd.merge(RFM,M,on="CUSTOMERNAME").sort_values(by="SALES",ascending=False)


RFM["DAYS_SINCE_LASTORDER"] = recency["Days_science_lastorder"]
RFM.rename(columns={"SALES":"MONETARY","ORDERNUMBER":"Frequency"},inplace=True)
RFM["Customers Statue"] = pd.qcut(RFM["MONETARY"],q=3,labels=["standard","Loyal","VIP"])


RFM['potential churn risk'] = 'At Risk'

RFM.loc[RFM['avg_purchase_interval'] == 0,'potential churn risk'] = 'One-Time Customer'

RFM.loc[
    (RFM['avg_purchase_interval'] > 0) &
    (RFM['DAYS_SINCE_LASTORDER'] <= RFM['avg_purchase_interval']),
    'potential churn risk'
] = 'Regular'


# Potential churn customers
RFM.loc[
    (RFM['avg_purchase_interval'] > 0) &
    (RFM['DAYS_SINCE_LASTORDER'] >= RFM['avg_purchase_interval'] * 2),
    'potential churn risk'
] = 'Potential Churn'

In [ ]:
pct_churn = (RFM[RFM["potential churn risk"]=="Potential Churn"]["potential churn risk"].count()/RFM.shape[0])*100
pct_one_time = (RFM[RFM["potential churn risk"]=="One-Time Customer"]["potential churn risk"].count()/RFM.shape[0])*100
pct_regular = (RFM[RFM["potential churn risk"]=="Regular"]["potential churn risk"].count()/RFM.shape[0])*100
pct_at_risk = (RFM[RFM["potential churn risk"]=="At Risk"]["potential churn risk"].count()/RFM.shape[0])*100
print(f"The Percentage of  churned One Time Customers: {round(pct_one_time,2)}%")

print(f"The Percentage of Regular Customers: {round(pct_regular,2)}%")
print(f"The Percentage of May be churned Customers: {round(pct_churn,2)}%")
print(f"The Percentage of  Customers at risk: {round(pct_at_risk,2)}%")
VIP_risk = RFM[(RFM["potential churn risk"]=="Potential Churn")&(RFM["Customers Statue"]== "VIP")]["Customers Statue"].count()/RFM.shape[0]*100
VIP_risk

The Percentage of  churned One Time Customers: 1.12%
The Percentage of Regular Customers: 56.18%
The Percentage of May be churned Customers: 21.35%
The Percentage of  Customers at risk: 21.35%


np.float64(3.3707865168539324)

In [ ]:
revenue_distribution  = RFM.groupby('Customers Statue')["MONETARY"].agg(["sum","count"]).reset_index()
revenue_distribution = revenue_distribution.rename(columns={"sum":"Monetary","count":"Customer number"})

revenue_distribution["Percentage of Revenue"] = (revenue_distribution["Monetary"]/company_revenue*100).round()
revenue_distribution.sort_values(by="Monetary",ascending=False)

/tmp/ipykernel_34778/2768601999.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  revenue_distribution  = RFM.groupby('Customers Statue')["MONETARY"].agg(["sum","count"]).reset_index()


,Customers Statue,Monetary,Customer number,Percentage of Revenue
2,VIP,5514483.62,30,56.0
1,Loyal,2602856.71,29,27.0
0,standard,1642881.38,30,17.0


# RFM analysis reveals customer segmentation along two main axes:

##Purchase Activity
* Most customers are repeat customers
* About 7% of customers are at risk of churn. Most of them are high-value customers with high revenue contribution.

* Approximately 21% of customers have not purchased for a long period, nearly double their usual average purchase interval. Most of these customers generate low revenue, which may indicate a high churn risk.

##Revenue Segmentation

Revenue is divided into three customer categories:

* VIP Customers
* Loyal Customers
* Standard Customers

VIP customers generate 56% of total revenue, while loyal customers contribute around 27%. The remaining percentage comes from standard customers.

This shows that a large part of revenue depends on VIP customers.

## Customer Recommendations
###At Risk Customers
* Conduct customer satisfaction surveys.
* Improve customer follow-up and communication.
###May Be Churned Customers
* Launch re-engagement campaigns.
* Understand customer behavior and concerns.
* Offer personalized promotions to encourage repeat purchases.
###Loyal and Standard Customers
* Create personalized offers and loyalty programs.
* Increase engagement through targeted marketing campaigns.
* VIP Customers

Provide exclusive services such as:

Early delivery
Faster shipping
Priority order fulfillment
Special customer experience services

# Geographic sales analysis

# Geographic sales analysis over time

In [ ]:
country_revenue = df_clean.groupby(["COUNTRY","ORDERDATE"])["SALES"].sum().reset_index(name="Revenue")
country_revenue["ORDERDATE"] = country_revenue["ORDERDATE"].dt.year
country_revenue = country_revenue.pivot_table(index="COUNTRY",columns="ORDERDATE",values="Revenue",aggfunc="sum")
country_revenue["2019 Growth"] = ((country_revenue[2019]-country_revenue[2018])/country_revenue[2018]*100).round()
country_revenue.fillna(0,inplace=True)

country_revenue["total revenue"] = country_revenue.sum(axis=1)
country_revenue["total revenue percentage"] = round((country_revenue["total revenue"]/company_revenue)*100)
country_revenue.sort_values(by="total revenue",ascending=False,inplace=True)
country_revenue.reset_index(inplace=True)

country_revenue.columns.name = None
country_revenue

,COUNTRY,2018,2019,2020,2019 Growth,total revenue,total revenue percentage
0,USA,1141182.40,1631232.65,583160.64,43.0,3355618.69,34.0
1,Spain,405343.39,483545.36,326798.17,19.0,1215705.92,12.0
2,France,312761.42,555198.70,242956.40,78.0,1110994.52,11.0
3,Australia,253134.45,232396.68,145091.97,-8.0,630615.10,6.0
4,UK,180421.55,257656.10,40802.81,43.0,478923.46,5.0
5,Italy,140928.77,192235.60,41509.94,36.0,374710.31,4.0
6,Finland,111154.51,91575.69,126851.71,-18.0,329563.91,3.0
7,Norway,196532.60,110931.10,0.00,-44.0,307419.70,3.0
8,Singapore,165686.20,116039.03,6763.18,-30.0,288458.41,3.0
9,Denmark,99192.72,120431.56,26012.87,21.0,245658.15,3.0


In [ ]:
bins = [-float('inf'), -1,0, 0.05, 0.15, float('inf')]
labels = ['Declining',"New entry", 'Low Growth', 'Stable Growth', 'High Growth']

country_revenue['Market_Priority'] = pd.cut(country_revenue['2019 Growth'], bins=bins, labels=labels)
country_revenue.sort_values(by="2019 Growth", ascending=False,inplace=True)
country_revenue

pec_of_growth_country = country_revenue[country_revenue["2019 Growth"]>15].shape[0]/country_revenue.shape[0]*100
pec_of_Declining_country = country_revenue[country_revenue["2019 Growth"]<0].shape[0]/country_revenue.shape[0]*100
pec_of_new_entry = country_revenue[country_revenue["2019 Growth"]==0].shape[0]/country_revenue.shape[0]*100
pec_of_Declining_country = country_revenue[country_revenue["2019 Growth"]<=-1].shape[0]/country_revenue.shape[0]*100
print(f"The Percentage of Countries with Low Growth: {round(pec_of_Declining_country)}%")
print(f"The Percentage of Countries with High Growth: {round(pec_of_growth_country)}%")
print(f"The Percentage of Countries with New Entry: {round(pec_of_new_entry)}%")
print(f"The Percentage of Countries with Declining Growth: {round(pec_of_Declining_country)}%")
country_revenue



In [ ]:
city_revenue = df_clean.groupby(["COUNTRY","CITY","ORDERDATE"])["SALES"].sum().reset_index(name="Revenue")
city_revenue["ORDERDATE"] = city_revenue["ORDERDATE"].dt.year
city_revenue = city_revenue.pivot_table(index=["COUNTRY","CITY"],columns="ORDERDATE",values="Revenue",aggfunc="sum").fillna(0)
city_revenue["total revenue"] = city_revenue.sum(axis=1)
city_revenue["total revenue percentage"] = round((city_revenue["total revenue"]/company_revenue)*100)
city_revenue.sort_values(by="total revenue",ascending=False,inplace=True)
city_revenue.reset_index(inplace=True)

city_revenue.columns.name = None
country_revenue
city_revenue

,COUNTRY,CITY,2018,2019,2020,total revenue,total revenue percentage
0,Spain,Madrid,305612.14,450141.13,326798.17,1082551.44,11.0
1,USA,San Rafael,185128.12,256474.26,213255.68,654858.06,7.0
2,USA,NYC,215488.21,345299.56,0.00,560787.77,6.0
3,Singapore,Singapore,165686.20,116039.03,6763.18,288488.41,3.0
4,France,Paris,63842.29,170714.06,34388.33,268944.68,3.0
...,...,...,...,...,...,...,...
66,Austria,Graz,43488.74,0.00,8775.16,52263.90,1.0
67,USA,Brisbane,34100.03,16118.48,0.00,50218.51,1.0
68,USA,Burbank,8234.56,37850.08,0.00,46084.64,0.0
69,Germany,Munich,0.00,34993.92,0.00,34993.92,0.0


In [ ]:
country_cities = city_revenue["COUNTRY"].value_counts().reset_index(name="City Number")
country_cities

,COUNTRY,City Number
0,USA,21
1,France,9
2,Australia,5
3,UK,4
4,Spain,3
5,Canada,3
6,Italy,3
7,Finland,3
8,Germany,3
9,Norway,3


In [ ]:
favor_products = df_clean.groupby(["COUNTRY","PRODUCTLINE"])["SALES"].sum().sort_values(ascending=False).reset_index()
favor_products[favor_products['PRODUCTLINE']=="Classic Cars"]
favor_products.pivot_table(index="COUNTRY",columns="PRODUCTLINE",values="SALES",aggfunc="sum").fillna(0).reset_index()

PRODUCTLINE,COUNTRY,Classic Cars,Motorcycles,Planes,Ships,Trains,Trucks and Buses,Vintage Cars
0,Australia,193085.54,89968.76,74853.87,4159.76,1681.35,77318.50,189555.32
1,Austria,101459.47,26047.66,17860.44,9024.73,0.00,20472.75,27197.48
2,Belgium,20136.96,0.00,5624.79,31708.01,9017.26,0.00,41925.60
3,Canada,61623.22,4177.49,25510.07,40309.01,0.00,51945.98,40512.79
4,Denmark,157182.48,0.00,7586.45,38697.26,11476.33,9588.82,21105.81
5,Finland,153552.24,47866.72,34375.13,29808.44,5117.05,40479.33,18383.00
6,France,388951.20,226390.31,108155.51,66486.67,27340.80,116982.22,176609.81
7,Germany,148315.00,7497.50,23001.26,5501.00,5043.42,10178.00,20935.91
8,Ireland,31688.82,4953.20,11784.36,0.00,3112.60,3983.05,2234.40
9,Italy,128576.65,7567.80,98185.65,17703.54,6274.96,5914.97,110450.74


#Geographic Analysis of Revenue (2018 – Mid 2020)
##Countries vs Revenue

* Three countries clearly dominate total revenue, representing approximately 57% of total company revenue.

* The United States ranks first in revenue and operates across 21 cities, followed by Spain with 3 cities, and France with 9 cities.

###Strengths

* These markets represent strong and important regions for the company and reflect customer trust in the brand and products.

### Risks

Revenue is concentrated in a limited number of countries. Therefore, any economic or political disruptions in these markets could significantly impact total company revenue.

##Countries vs Growth (Based on 2019 Growth Due to Incomplete 2020 Data)

* Approximately 53% of markets achieved strong growth during 2019.

* Around 32% of markets showed declining performance.

* In addition, several markets recorded revenue activity in 2019 despite having no recorded revenue in 2018, which may indicate expansion into newer markets.

##Revenue by City

* Three cities contribute approximately 24% of total revenue.

* Madrid ranks first in total revenue, followed by San Rafael, then NYC.

##Recommendations
###High-Revenue Markets

* High-performing markets should receive larger marketing investments and exclusive business advantages, such as launching new products earlier in these regions.

* Regional managers may also be assigned to oversee these important markets more effectively.

###Growing Markets

* Growing markets should be monitored closely and supported with additional marketing campaigns and sales expansion strategies.

###Declining Markets

* Declining markets should be reviewed carefully to better understand the reasons behind weaker performance, including customer behavior, competition, pricing, and market conditions.

* Additional customer engagement campaigns and after-sales support may help improve performance.

### Customer Support & Market Understanding

* Customer interests and purchasing behavior should be studied more deeply in order to provide better technical support, after-sales services, and more personalized offerings.

# Executive Summary

#KPIs
 Throug uring the period from 2018 to mid-2020

* Total revenue: 9,760,221,71M
* Total Orders: 298 order
* Total Customers: 89
* Top products Category : Classic Cars,Vintage Cars
* Top products Category Contributions:(39,19) 58%
* Top Market: USA
* Top Market Contribution : 34%
* Customers at Risk of Churn: 24%
* Top City Contribution: 11%
* Top Declining Market : Philippines

# Key Business Insights
## Revenue Trends
* Revenue showed strong growth during the second half of the year, especially during Q4, indicating seasonal purchasing behavior.

## Customer Behavior
* Most revenue is generated by repeat customers, which reflects strong customer loyalty and satisfaction.

* However, a small percentage of high-value customers show reduced purchasing activity and may be at risk of churn.

##Product Performance
* Classic Cars, vintage Cars , Trucks and Busses products generate The highest the highest revenue contribution and Showed consistently strong demand across all regions

* Train products saw a significant drop in profits

##Geograpic performance

* Revenue is concentrated in a limited number of regions, with two major cities contributing nearly half of total sales.

*Several emerging markets demonstrated strong year-over-year growth and may represent future expansion opportunities.

##Business Risks
* Heavy dependence on a small number of regions
* Potential churn among high-value customers
* Weak performance in certain product categories

##Recommendations
###Customer Retention

* Launch loyalty campaigns and personalized offers targeting high-value customers.

###Market Expansion

* Increase marketing investments in fast-growing regions and monitor emerging markets closely.

* Product Strategy

Focus on expanding high-performing categories while reviewing pricing and demand for low-performing products.

##Operations

* Improve customer support and delivery performance during high-demand seasonal periods.

In [ ]:
RFM.to_csv("RFM.csv", index=False)

# Key Business Insights

- Q4 generates the highest revenue across all periods.
- VIP customers contribute the majority of total revenue.
- Revenue is concentrated among a limited number of product lines.
- Geographic sales performance is concentrated in a few key markets.


# Business Recommendations

- Focus marketing campaigns during Q4 peak periods.
- Retain high-value VIP customers.
- Re-engage inactive customers.
- Expand operations in high-performing markets.


# Conclusion

The analysis identified strong seasonal revenue patterns, customer concentration risks, and valuable customer segmentation insights that can support future business growth and customer retention strategies.
